lib/AST/Type.cpp

```diff
@@ -906,9 +906,13 @@ Type TypeBase::lookThroughAllOptionalTypes(SmallVectorImpl<Type> &optionals){
 }
 
 unsigned int TypeBase::getOptionalityDepth() {
-  SmallVector<Type> types;
-  lookThroughAllOptionalTypes(types);
-  return types.size();
+  Type type(this);
+  unsigned int depth = 0;
+  while (auto objType = type->getOptionalObjectType()) {
+    type = objType;
+    ++depth;
+  }
+  return depth;
```

Comment:
I think we should keep this the way it was really or change it in a separate commit.

Reply:
This was in a different commit than actual changes already, but if you think is worth another NFC PR for that I can remove and open separately.

Reply:
Let's land this separately because it's not technically necessary for the fix since the resulting count is the same.

Reply:
Sure, I'll land it separately =]